### PR TITLE
Api tests table

### PR DESCRIPTION
**Pull request by ahuzmeza**

### CODE REVIEW

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift [hunk 1]
```diff
@@ -12,8 +12,8 @@ final class CoursesViewController: UIViewController,
                                    UICollectionViewDataSource,
                                    UICollectionViewDelegateFlowLayout {
     @IBOutlet private var courseCollectionView: UICollectionView!
-    private var courseNames: [String] = ["Geography", "Mathematics", "Biology", "Chemistry", "Informatics"]
-    private var courseIcons: [String] = ["Geography", "Mathematics", "Biology", "Chemistry", "Informatics"]
+    
+    var coursesData: [CoursesModel] = []
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift]: make it private

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift [hunk 2]
```diff
@@ -24,6 +24,20 @@ final class CoursesViewController: UIViewController,
         courseCollectionView.register(nibCell, forCellWithReuseIdentifier: "CoursesCollectionViewCell")
     }
     
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(true)
+        getCourses()
+    }
+    
+    func getCourses() {
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift]: make it private

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift [hunk 3]
```diff
@@ -21,15 +21,13 @@ class TestsViewController: UIViewController {
         case passedButton
     }
     private var currentButton = NavButtons.chaptersButton
-
+    
+    var course: CoursesModel?
     // MARK: - DataSet for 'chapters', 'final' 'passed' tests
-    var chaptersDataSource: [String] = []
+    var chaptersDataSource: [ChapterModel] = []
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift]: make it private

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift [hunk 4]
```diff
@@ -21,15 +21,13 @@ class TestsViewController: UIViewController {
...
     var passedDataSource: [String] = []
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift]: make it private

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift [hunk 5]
```diff
@@ -21,15 +21,13 @@ class TestsViewController: UIViewController {
...
-    let finalDataSource: [FinalTestModel] = [
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift:28]: make it private

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift [hunk 6]
```diff
@@ -192,4 +194,12 @@ extension TestsViewController: UITableViewDataSource, UITableViewDelegate {
             return 320
         }
     }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let viewController = QuizViewController()
+        viewController.modalPresentationStyle = .fullScreen
+        viewController.navigationItem.title = chaptersDataSource[indexPath.row].name
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/TestsSegmentedController/TestsViewController.swift]: set title in QuizViewController, not here

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift [hunk 7]
```diff
@@ -34,15 +48,25 @@ final class CoursesViewController: UIViewController,
...
         }
         return UICollectionViewCell()
     }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let coursesData = CoursesApiManager.sharedCoursesData.courses
+        
+        let viewController = TestsViewController()
+        viewController.modalPresentationStyle = .fullScreen
+        viewController.navigationItem.title = coursesData[indexPath.row].name
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift]: set title in TestsViewController, not here

**Review by iOSinternshipen — CHANGES_REQUESTED** [review]

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift (hunk 1 above)
Comment (ahuzmeza) [reply Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift]: fixed after merge with develop

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift (hunk 2 above)
Comment (ahuzmeza) [reply Internship2022TeamSzabi/Internship2022TeamSzabi/Views/CoursesViewController/CoursesViewController.swift]: fixed after merge with develop

**Pull request merged**